gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```
@@ -82,13 +83,30 @@ public Optional<ProxyBackendConfiguration> getBackendByName(String name)
     @Override
     public void deactivateBackend(String backendName)
     {
-        dao.deactivate(backendName);
+        updateBackendActivationStatus(backendName, () -> dao.deactivate(backendName));
     }
 
     @Override
     public void activateBackend(String backendName)
     {
-        dao.activate(backendName);
+        updateBackendActivationStatus(backendName, () -> dao.activate(backendName));
+    }
+
+    private void updateBackendActivationStatus(String backendName, Runnable action)
+    {
+        GatewayBackend model = dao.findFirstByName(backendName);
+        if (model == null) {
+            throw new IllegalStateException("No backend found with name: " + backendName + ", could not (de)activate");
+        }
+
+        Boolean prevStatus = model.active();
+        action.run();
+        Boolean currStatus = dao.findFirstByName(backendName).active();
+
+        if (!Objects.equals(prevStatus, currStatus)) {
```

IMO this check is redundant as the function is "updating" the status. If prevStatus and currStatus is the same for some reason, then it should be db level failure, in any case we should throw an exception

---

Rather than a db level failure, what if someone just calls /activate on an already active backend? This check prevents us from unnecessarily logging in this case. Also, it keeps it consistent with logging in updateBackend which does require checking if prevStatus != currStatus, because changes to activation status made via the UI use updateBackend, but not all calls to updateBackend method change the activation status, so we need to check if a change has been made to log it.